test/e2e/query_test.go

```diff
@@ -368,31 +368,35 @@ func checkNetworkRequests(t *testing.T, addr string) {
 	ctx, cancel := chromedp.NewContext(context.Background())
 	t.Cleanup(cancel)
 
-	var networkErrors []string
+	testutil.Ok(t, runutil.Retry(1*time.Minute, ctx.Done(), func() error {
+		var networkErrors []string
 
-	// Listen for failed network requests and push them to an array.
-	chromedp.ListenTarget(ctx, func(ev interface{}) {
-		switch ev := ev.(type) {
-		case *network.EventLoadingFailed:
-			networkErrors = append(networkErrors, ev.ErrorText)
-		}
-	})
-
-	err := chromedp.Run(ctx,
-		network.Enable(),
-		chromedp.Navigate(addr),
-		chromedp.WaitVisible(`body`),
-	)
-	testutil.Ok(t, err)
+		err := chromedp.Run(ctx,
+			network.Enable(),
+			chromedp.Navigate(addr),
+			chromedp.WaitVisible(`body`),
+		)
 
-	err = func() error {
-		if len(networkErrors) > 0 {
-			return fmt.Errorf("some network requests failed: %s", strings.Join(networkErrors, "; "))
+		if err != nil {
+			return err
 		}
-		return nil
-	}()
 
-	testutil.Ok(t, err)
+		// Listen for failed network requests and push them to an array.
+		chromedp.ListenTarget(ctx, func(ev interface{}) {
+			switch ev := ev.(type) {
+			case *network.EventLoadingFailed:
+				networkErrors = append(networkErrors, ev.ErrorText)
+			}
+		})
+
+		err = func() error {
```

Comment:
Is there a reason why this is a function? :s Does it give any benefits? I think we can move all of the code inside of the function to the outside.

Reply:
No, there are no benefits.